ruby/ext/google/protobuf_c/repeated_field.c

```diff
@@ -391,44 +375,6 @@ VALUE RepeatedField_eq(VALUE _self, VALUE _other) {
   return Qtrue;
 }
 
-/*
- * call-seq:
- *     RepeatedField.hash => hash_value
- *
- * Returns a hash value computed from this repeated field's elements.
- */
-VALUE RepeatedField_hash(VALUE _self) {
-  RepeatedField* self = ruby_to_RepeatedField(_self);
-
-  VALUE hash = LL2NUM(0);
-
-  upb_fieldtype_t field_type = self->field_type;
-  VALUE field_type_class = self->field_type_class;
-  size_t elem_size = native_slot_size(field_type);
-  size_t off = 0;
-  for (int i = 0; i < self->size; i++, off += elem_size) {
-    void* mem = ((uint8_t *)self->elements) + off;
-    VALUE elem = native_slot_get(field_type, field_type_class, mem);
-    hash = rb_funcall(hash, rb_intern("<<"), 1, INT2NUM(2));
-    hash = rb_funcall(hash, rb_intern("^"), 1,
-                      rb_funcall(elem, rb_intern("hash"), 0));
-  }
-
-  return hash;
-}
```

Comment:
I removed this because it wasn't equivalent to the ruby version.  I'm not totally sure what sort of impact that has in the ruby ecosystem... I highly doubt someone would be using a `RepeatedField` as the key in a ruby hash!  But they may in a Ruby Set, perhaps?  The performance is almost exactly a 2x delta.